src/server/cluster/cluster_config.h

```diff
@@ -3,36 +3,83 @@
 //
 
 #pragma once
-#include <absl/container/flat_hash_set.h>
 
+#include <absl/base/thread_annotations.h>
+
+#include <array>
+#include <optional>
 #include <string_view>
-#include <tuple>
+#include <vector>
 
 #include "src/core/fibers.h"
 
 namespace dfly {
 
-typedef uint16_t SlotId;
+using SlotId = uint16_t;
 
 class ClusterConfig {
  public:
-  ClusterConfig();
+  struct Node {
+    std::string id;
+    std::string ip;
+    uint16_t port = 0;
+  };
+
+  struct SlotRange {
+    SlotId start = 0;
+    SlotId end = 0;
+  };
+
+  struct ClusterShard {
+    std::vector<SlotRange> slot_ranges;
+    Node master;
+    std::vector<Node> replicas;
+  };
+
+  explicit ClusterConfig(std::string_view my_id);
+
   static SlotId KeySlot(std::string_view key);
+
   static bool IsClusterEnabled() {
     return cluster_enabled;
   }
+
   // If the key contains the {...} pattern, return only the part between { and }
   static std::string_view KeyTag(std::string_view key);
 
   // If key is in my slots ownership return true
-  bool IsMySlot(SlotId id);
+  bool IsMySlot(SlotId id) const;
+
+  // Returns nodes that own `id`. Result will always have the first element set to be the master,
+  // and the following 0 or more elements are the replicas.
+  std::vector<Node> GetNodesForSlot(SlotId id) const;
```

Comment:
what is the flow we will need this api for?

Reply:
`CLUSTER SHARDS` for example

Reply:
Removed per the new API